src/Home/hooks/useDeleteFavorite.ts

```diff
@@ -0,0 +1,18 @@
+import { useMutation } from 'react-query';
+import { useNavigate } from 'react-router-dom';
+
+import deleteFavorite from '../api/deleteFavorite';
+
+const useDeleteFavorite = () => {
+  const navigate = useNavigate();
+  const mutation = useMutation({
+    mutationFn: (bookId: number) => {
+      return deleteFavorite(bookId);
+    },
+    onError: () => navigate('/error'),
+    onSuccess: () => location.reload(),
```

Comment:
```suggestion
  onSuccess: () => {
  //  const queryClient = useQueryClient(); Import 해주세용
    queryClient.refetchQueries(['get-favorite'], { exact: true });
  },
```
deleteFavorite 함수가 성공적으로 실행되면, 해당 쿼리키를 가진 쿼리를 다시 실행 시키는 함수입니다!

 exact 옵션으로 정확하게 일치하는 쿼리 키를 가진 쿼리만 다시 실행하도록 한다고해용

 자연스럽게 데이터가 변화하니까 되니 한 번 고려 해보아도 좋겟어요!!!